pandas/tests/indexes/multi/test_sorting.py

```diff
@@ -271,3 +271,24 @@ def test_argsort(idx):
     result = idx.argsort()
     expected = idx.values.argsort()
     tm.assert_numpy_array_equal(result, expected)
+
+
+def test_remove_unused_levels_with_missing():
+    # GH 37510
+    df1 = DataFrame(
+        {
+            "L1": [1, 2, 3, 4],
+            "L2": [3, 4, 1, 2],
+            "L3": [1, 1, 1, 1],
+            "x": [1, 2, 3, 4],
+        }
+    )
+    df1 = df1.set_index(["L1", "L2", "L3"])
+    new_levels = ["n1", "n2", "n3", None]
+    df1.index = df1.index.set_levels(levels=new_levels, level="L1")
+    df1.index = df1.index.set_levels(levels=new_levels, level="L2")
+
+    result = df1.unstack("L3")[("x", 1)].sort_index().index
+    expected = df1.droplevel(2, 0).sort_index().index
```

Comment:
can you hardcode the expected frame entirely here (not averse to *also* doing it programatically)